documentation/quarkus/guide-beanmapping-quarkus.asciidoc

```diff
@@ -0,0 +1,224 @@
+  ... 
+}
+----
+
+=== MapperConfig Configuration
+Create a shared configuration that can be used for multiple mappers. Implement an Interface and use the annotation *@MapperConfig* for the class. You can define all configurations in this interface and pass the generated `MapperConfig.class` with the *config* attribute to the mapper. The `MapperConfig` also defines the *InjectionStrategy* and *MappingInheritaceStrategy* both will be explained later.
+A list of all configurations can be found https://mapstruct.org/documentation/stable/api/org/mapstruct/MapperConfig.html[here].
+[source, java]
+----
+@MapperConfig(
+  compnentModel = "cdi",
+  mappingInheritanceStrategy = MappingInheritanceStrategy.AUTO_INHERIT_FROM_CONFIG
+  injectionStrategy =InjectionStrategy.CONSTRUCTOR
+)
+public interface MapperConfig{
+}
+----
+
+[source, java]
+----
+@Mapper( config = MapperConfig.class )
+public interface AnimalMapper{
+  ...
+}
+----
+Any attributes notgiven via *@Mapper* will be inherited from the shared configuration `MapperConfig.class`.
+
+=== Configuration via annotaion processor options
+The MapStruct code generator can be configured using annotation processor options.
+You can pass the options to the compiler while invoking javac directly, or add the parameters to the maven configuration in the POM.xml
+[source, xml]
+----
+<build>
+  <plugins>
+    <plugin>
+      <artifactId>maven-compiler-plugin</artifactId>
+      <version>${compiler-plugin.version}</version>
+        <configuration>
+          <source>1.8</source>
+          <target>1.8</target>
+          <annotationProcessorPaths>
+            <path>
+                <groupId>org.mapstruct</groupId>
+                <artifactId>mapstruct-processor</artifactId>
+                <version>${org.mapstruct.version}</version>
+            </path>
+          </annotationProcessorPaths>
+        <parameters>true</parameters>
+        <compilerArgs>
+          <!-- mapstruct, please generate all classes as CDI beans -->
+          <arg>-Amapstruct.defaultComponentModel=cdi</arg>
+          <arg>-Amapstruct.defaultInjectionStrategy=CONSTRUCTOR</arg>
+        </compilerArgs>
+      </configuration>
+    </plugin>
+  </plugins>
+</build>
+----
+A list of all annotation processor options can be found https://mapstruct.org/documentation/dev/reference/html/#configuration-options[here]. 
+
+We are also using the constructor injection strategie to avoid field injections and potential reflections also it will simplify our tests.
+The option to pass the parameter to the annotation processor in the POM.xml is used and can be inspected in our https://github.com/devonfw-sample/devon4quarkus-reference[quarkus reference project].
+
+
+
+
+== Basic Bean-Mapper Usage
+
+To use the mapper we have to implement the mapper interface and the function prototypes with a *@Mapper* annotation. 
+
+[source, java]
+----
+@Mapper
+public interface AnimalMapper {
+
+  AnimalDto animal2AnimalDto(Animal entity);
```

Comment:
Is this a common notation for naming the methods? In out reference project we simply use `map` or `create`

Reply:
Took the names from the mapstruct guide, i didn´t want to use two times map for a mapping function and the inverse of it.